--- libutempter-1.1.6/utempter.c.orig	2010-11-04 18:14:53.000000000 +0100
+++ libutempter-1.1.6/utempter.c	2012-06-23 21:51:40.962552952 +0200
@@ -113,20 +113,27 @@
 {
 	struct utmp ut;
 	struct timeval tv;
+	int fd;
+	struct lastlog ll;
 
 #ifdef __GLIBC__
 	size_t offset;
 #endif
 
 	memset(&ut, 0, sizeof(ut));
+	memset(&ll, 0, sizeof(ll));
 
 	memset(&tv, 0, sizeof(tv));
 	(void) gettimeofday(&tv, 0);
+	ll.ll_time = time(NULL);
 
 	strncpy(ut.ut_name, user, sizeof(ut.ut_name));
 	strncpy(ut.ut_line, term, sizeof(ut.ut_line));
-	if (host)
+	strncpy(ll.ll_line, term, sizeof(ll.ll_line));
+	if (host) {
 		strncpy(ut.ut_host, host, sizeof(ut.ut_host));
+		strncpy(ll.ll_host, host, sizeof(ll.ll_host));
+	}
 
 #ifdef __GLIBC__
 
@@ -155,6 +162,11 @@
 	endutent();
 
 	(void) updwtmp(_PATH_WTMP, &ut);
+	if ((fd = open(_PATH_LASTLOG, O_RDWR)) != -1) {
+		lseek(fd, (off_t) ((long)getuid() * sizeof(ll)), SEEK_SET);
+		write(fd, &ll, sizeof(ll));
+		close(fd);
+	}
 
 #elif defined(__FreeBSD__)
 
