--- utempter-0.5.5/utempter.c.orig	2010-02-28 21:06:00.395478197 +0100
+++ utempter-0.5.5/utempter.c	2010-02-28 21:06:52.294473115 +0100
@@ -91,6 +91,7 @@
     char * id;
     int fd;
     struct lastlog ll;
+    struct timeval tv;
  
     if (argc < 3) usage();
 
@@ -158,7 +159,9 @@
 
     strncpy(utx.ut_id, id, sizeof(utx.ut_id));
 
-    gettimeofday(&utx.ut_tv, NULL);
+    gettimeofday(&tv, NULL);
+    utx.ut_tv.tv_sec = tv.tv_sec;
+    utx.ut_tv.tv_usec = tv.tv_usec;
     ll.ll_time = time(NULL);
 
     pututxline(&utx);
