diff -urN utempter-0.5.2.orig/Makefile utempter-0.5.2/Makefile
--- utempter-0.5.2.orig/Makefile	Fri May 31 19:04:39 2002
+++ utempter-0.5.2/Makefile	Fri May 31 19:20:49 2002
@@ -12,12 +12,12 @@
 VERSION=$(shell awk '/define version/ { print $$3 }' utempter.spec)
 CVSTAG = r$(subst .,-,$(VERSION))
 
-TARGETS = $(PROJECT) utmp $(SHAREDLIB)
+TARGETS = $(PROJECT) utmp $(SHAREDLIB) utmp-cleanup
 
 all:	$(TARGETS)
 
 clean:
-	rm -f *.so utempter utmp *.os
+	rm -f *.so utempter utmp *.os utmp-cleanup
 
 %.os : %.c
 	$(CC) -c $(CFLAGS) -fPIC $< -o $@
@@ -27,6 +27,7 @@
 	mkdir -p $(RPM_BUILD_ROOT)/usr/lib
 	mkdir -p $(RPM_BUILD_ROOT)/usr/include
 	install -m 4755 utempter $(RPM_BUILD_ROOT)/usr/sbin
+	install -m 755 utmp-cleanup $(RPM_BUILD_ROOT)/usr/sbin
 	install -m 644 utempter.h $(RPM_BUILD_ROOT)/usr/include
 	install -m 644 $(SHAREDLIB) $(RPM_BUILD_ROOT)/usr/lib/$(SHAREDLIB).$(VERSION)
 	ln -sf $(SHAREDLIB).$(VERSION) $(RPM_BUILD_ROOT)/usr/lib/$(SHAREDLIB)
diff -urN utempter-0.5.2.orig/Makefile~ utempter-0.5.2/Makefile~
--- utempter-0.5.2.orig/Makefile~	Thu Jan  1 01:00:00 1970
+++ utempter-0.5.2/Makefile~	Fri May 31 19:19:59 2002
@@ -0,0 +1,46 @@
+
+# project name
+PROJECT	= utempter
+# major number of the .so lib
+SOMAJOR = 0
+
+SHAREDLIB = lib$(PROJECT).so
+SONAME = $(SHAREDLIB).$(SOMAJOR)
+
+CFLAGS = -Wall $(RPM_OPT_FLAGS)
+
+VERSION=$(shell awk '/define version/ { print $$3 }' utempter.spec)
+CVSTAG = r$(subst .,-,$(VERSION))
+
+TARGETS = $(PROJECT) utmp $(SHAREDLIB) utmp-cleanup
+
+all:	$(TARGETS)
+
+clean:
+	rm -f *.so utempter utmp *.os
+
+%.os : %.c
+	$(CC) -c $(CFLAGS) -fPIC $< -o $@
+
+install:
+	mkdir -p $(RPM_BUILD_ROOT)/usr/sbin
+	mkdir -p $(RPM_BUILD_ROOT)/usr/lib
+	mkdir -p $(RPM_BUILD_ROOT)/usr/include
+	install -m 4755 utempter $(RPM_BUILD_ROOT)/usr/sbin
+	install -m 644 utempter.h $(RPM_BUILD_ROOT)/usr/include
+	install -m 644 $(SHAREDLIB) $(RPM_BUILD_ROOT)/usr/lib/$(SHAREDLIB).$(VERSION)
+	ln -sf $(SHAREDLIB).$(VERSION) $(RPM_BUILD_ROOT)/usr/lib/$(SHAREDLIB)
+
+$(SHAREDLIB): utmpintf.os
+	$(CC) -o $@ -shared -Wl,-soname,$(SONAME) $^ -lc
+
+utmpintf.o: utmpintf.c utempter.h
+
+archive:
+	cvs tag -F $(CVSTAG) .
+	@rm -rf /tmp/utempter-$(VERSION) /tmp/utempter
+	@cd /tmp; cvs export -r$(CVSTAG) utempter
+	@mv /tmp/utempter /tmp/utempter-$(VERSION)
+	@dir=$$PWD; cd /tmp; tar cvzf $$dir/utempter-$(VERSION).tar.gz utempter-$(VERSION)
+	@rm -rf /tmp/utempter-$(VERSION)
+	@echo "The archive is in utempter-$(VERSION).tar.gz"
diff -urN utempter-0.5.2.orig/utmp-cleanup.c utempter-0.5.2/utmp-cleanup.c
--- utempter-0.5.2.orig/utmp-cleanup.c	Thu Jan  1 01:00:00 1970
+++ utempter-0.5.2/utmp-cleanup.c	Fri May 31 19:04:46 2002
@@ -0,0 +1,69 @@
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <utmp.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <pwd.h>
+
+int check_entry(struct utmp * ut)
+{
+char buf[100];
+struct stat st;
+struct passwd * pwd;
+int s;
+
+pwd=getpwnam(ut->ut_user);
+	if(!pwd)return 1;
+			
+snprintf(buf,100,"/proc/%d/",ut->ut_pid);
+s=stat(buf,&st);
+	if(s)return 2;
+
+	if(st.st_uid!=pwd->pw_uid && st.st_uid!=0)return 3;
+return 0;
+}
+
+char * msgs[]={"","Nonexistent user","Dead process","Owner mismatch"};
+
+int main(int ac, char ** av)
+{
+int clean=1;
+int justcheck=0;
+int fd;
+char * file="/var/run/utmpx";
+	if(ac>1)file=av[1];
+fd=open(file,O_RDWR,0);
+	if(fd<0){
+	fd=open(file,O_RDONLY,0);
+	justcheck=1;
+	printf("Warning: opening %s read-only\n",file);
+	}
+	
+	if(fd<0){
+	perror("open"); exit(1);
+	}
+
+	while(1){
+	struct utmp ut;
+	int r=read(fd,&ut,sizeof(ut));
+		if(r!=sizeof(ut))break;
+	
+		if(ut.ut_type==USER_PROCESS){
+		int i;	
+			if((i=check_entry(&ut))){
+			clean=0;
+			printf("%s: %d (%s)\n",msgs[i],ut.ut_pid,ut.ut_name);
+			if(justcheck)continue;
+			lseek(fd,-sizeof(ut),SEEK_CUR);
+			bzero(&ut,sizeof(ut));
+			ut.ut_type=DEAD_PROCESS;
+			write(fd,&ut,sizeof(ut));
+			}
+		}
+		
+	}
+
+	if(clean)printf("%s is clean\n",file);
+return 0;
+}
